tls/s2n_cipher_suites.c

```diff
@@ -91,6 +91,11 @@ int s2n_set_cipher_as_client(struct s2n_connection *conn, uint8_t wire[S2N_TLS_C
         S2N_ERROR(S2N_ERR_CIPHER_NOT_SUPPORTED);
     }
 
+    /* Server gave us a cipher this version shouldn't support */
+    if (conn->pending.cipher_suite->minimum_required_tls_version > conn->actual_protocol_version) {
+        S2N_ERROR(S2N_ERR_CIPHER_NOT_SUPPORTED);
+    }
+
```

Comment:
In this case though, why not try to continue? if it works, great! if not, there's no other option.